vcn_gateways.tf

```diff
@@ -210,3 +210,21 @@ resource "oci_core_drg_attachment" "drg" {
 
   count = var.create_drg == true ? 1 : 0
 }
+
```

Comment:
I suppose LPG support would be incomplte without internet and nad gateway rules, same as with DRG. So I suggest a code like this for **oci_core_route_table.nat** and **oci_core_route_table.ig**
```terraform
  dynamic "route_rules" {
    # * filter var.nat_gateway_route_rules for routes with "lpg[lpg_KEY]" as destination
    # * and steer traffic to the module created LPG with given key
    for_each = var.nat_gateway_route_rules != null ? { for k, v in var.nat_gateway_route_rules : k => v
    if v.network_entity_id == "lpg" } : {}

    content {
      destination       = route_rules.value.destination
      destination_type  = route_rules.value.destination_type
      network_entity_id = oci_core_local_peering_gateway.lpg[route_rules.value.lpg_key].id
      description       = route_rules.value.description
    }
  }
```

Reply:
Hello Andrey,

Thanks for bringing this point to my attention.

The idea here is to focus on a simple task (create LPGs) and to be as close possible to the provider capability. What's important, is to be able to attach (any) route table, whether it is managed by the module or not.

For the content of the attached route table, yes it is critical to be able to control it, but that's more related to the route tables custom routes feature and should be kept out of this PR. In the near future, the vcn module will also introduce full capability to create/manage route tables, not only the `internet_route` and `nat_route` route tables automatically created with the gateways.

I would suggest you create a separate PR to add LPG as a valid destination for custom_route_rules.

Reply:
Okay, I may be will open a PR, cause I have the core code. I don't recall any use cases of LPG without subnet routing tables support and the routing table attached to LPG can only route to DRG. Maybe I'm missing some knowledge here...

Reply:
You are correct, if traffic is not routed from subnets to LPG nor it is used with a DRG, there is no use for this LPG.

Creating a route table and modifying its content may or may not be something you want to be managed by this module or even by Terraform.

Here, it is only about breaking down the end-to-end use cases (hub-and-spoke, 1:1 vcn peering) into smaller problems. First, allow creation of one or many LPG, then iterate to solve the adjacent problems (how do we route traffic to the LPG, etc ...).

Reply:
+1 for submitting this as a separate pr